x/upgrade/abci_test.go

```diff
@@ -38,6 +41,7 @@ func (s *TestSuite) SetupTest() {
 	s.cms.MountStoreWithDB(key, sdk.StoreTypeIAVL, db)
 	_ = s.cms.LoadLatestVersion()
 	s.ctx = sdk.NewContext(s.cms, abci.Header{Height: 10, Time: time.Now()}, false, log.NewNopLogger())
+	s.FlagUnsafeSkipUpgrade = FlagUnsafeSkipUpgrade
```

Comment:
And then do `viper.Set(FlagUnsafeSkipUpgrade, s.skipUpgrade)` here, rather than inside the tests code.

Otherwise, you can remove the calls to `module.BeginBlock` and just call the BeginBlocker directly and avoid all use of viper, which I think is cleaner. I actually only use the module here to try to increase the test coverage a bit more. Not sure if it makes sense to keep using it for BeginBlock or just call the function directly.

Reply:
In any case, this should be set for all calls in the Setup, not in individual tests, where it is easy to forget

Reply:
This(viper.set) should make it cleaner,will change.
I think we need to check the flag is being read in BeginBlock rather than just testing BeginBlocker

Reply:
Fair enough

Reply:
Okay